.github/bin/merge_rust_coverage.py

```diff
@@ -0,0 +1,100 @@
+# This file is dual licensed under the terms of the Apache License, Version
+# 2.0, and the BSD License. See the LICENSE file in the root of this repository
+# for complete details.
+
+import collections
+import sys
+
+import coverage
+
+
+class RustCoveragePlugin(coverage.CoveragePlugin):
+    def file_reporter(self, filename: str):
+        return RustCoverageFileReporter(filename)
+
+
+class RustCoverageFileReporter(coverage.FileReporter):
+    def lines(self) -> set[int]:
+        # XXX: Need a better way to handle this state!
+        return set(raw_data[self.filename])
+
+    def arcs(self) -> set[tuple[int, int]]:
+        return {(-1, line) for line in self.lines()}
+
+
+def coverage_init(
+    reg: coverage.plugin_support.Plugins,
+    options: coverage.types.TConfigSectionOut,
+) -> None:
+    reg.add_file_tracer(RustCoveragePlugin())
+
+
+def main(*lcov_paths: str):
+    cov = coverage.Coverage()
+    # XXX: Nasty mucking in semi-public APIs
```

Comment:
We don't need to do this if https://github.com/nedbat/coveragepy/pull/1919 or something like it merges right?

Reply:
Correct